ntp-daemon/src/peer.rs

```diff
@@ -288,7 +288,7 @@ where
     ) -> tokio::task::JoinHandle<()> {
         tokio::spawn(async move {
             let socket = loop {
-                match UdpSocket::new("0.0.0.0:0", &addr).await {
+                match UdpSocket::new("0.0.0.0:0", Some(&addr)).await {
```

Comment:
suggestion: have two constructors: one that takes the addr, and one that doesn't, with clear names